lib/facter/java_libjvm.rb

```diff
@@ -0,0 +1,32 @@
+# Fact: java_libjvm
+#
+# Purpose: get path to libjvm.so
+#
+# Resolution:
+#   Lists file in java package and searches for the file
+#
+# Caveats:
+#   Needs to query the package database
+#
+# Notes:
+#   None
+Facter.add(:java_libjvm) do
+  confine :osfamily => "RedHat"
+  setcode do
+    if Facter::Util::Resolution.which('rpm')
+      java_path = Facter.value(:java_path)
+      Facter::Util::Resolution.exec('rpm -qf '+java_path+' -l | grep libjvm.so\$').lines.first.strip
```

Comment:
See also my comment below about using string interpolation instead of concatenation.